helios-core/src/main/kotlin/helios/core/monoid.kt

```diff
@@ -0,0 +1,104 @@
+    override fun JsLong.combine(b: JsLong): JsLong = JsLong(Long.semigroup().run {
+        value.combine(b.value)
+    })
+}
+
+interface JsLongMonoid : Monoid<JsLong>, JsLongSemigroup {
+    override fun empty(): JsLong = JsLong(0L)
+}
+
+interface JsFloatSemigroup : Semigroup<JsFloat> {
+    override fun JsFloat.combine(b: JsFloat): JsFloat = JsFloat(Float.semigroup().run {
+        value.combine(b.value)
+    })
+}
+
+interface JsFloatMonoid : Monoid<JsFloat>, JsFloatSemigroup {
+    override fun empty(): JsFloat = JsFloat(0f)
+}
+
+interface JsDoubleSemigroup : Semigroup<JsDouble> {
+    override fun JsDouble.combine(b: JsDouble): JsDouble = JsDouble(Double.semigroup().run {
+        value.combine(b.value)
+    })
+}
+
+interface JsDoubleMonoid : Monoid<JsDouble>, JsDoubleSemigroup {
+    override fun empty(): JsDouble = JsDouble(0.0)
+}
+
+interface JsStringSemigroup : Semigroup<JsString> {
+    override fun JsString.combine(b: JsString): JsString = JsString(String.semigroup().run {
+        value.toString().combine(b.value.toString())
+    })
+}
+
+interface JsStringMonoid : Monoid<JsString>, JsStringSemigroup {
+    override fun empty(): JsString = JsString("")
+}
+
+interface JsArraySemigroup : Semigroup<JsArray> {
+    override fun JsArray.combine(b: JsArray): JsArray = JsArray(ListK.semigroup<Json>().run {
+        value.k().combine(b.value.k())
+    })
+}
+
+interface JsArrayMonoid : Monoid<JsArray>, JsArraySemigroup {
+    override fun empty(): JsArray = JsArray(emptyList())
+}
+
+interface JsObjectSemigroup : Semigroup<JsObject> {
+    override fun JsObject.combine(b: JsObject): JsObject = JsObject(value + b.value)
+}
+
+interface JsObjectMonoid : Monoid<JsObject>, JsObjectSemigroup {
+    override fun empty(): JsObject = JsObject(emptyMap())
+}
+
+fun JsInt.Companion.semigroup(): JsIntSemigroup = object : JsIntSemigroup {  }
+
+fun JsInt.Companion.monoid(): JsIntMonoid = object : JsIntMonoid {  }
+
+fun JsLong.Companion.semigroup(): JsLongSemigroup = object : JsLongSemigroup {  }
+
+fun JsLong.Companion.monoid(): JsLongMonoid = object : JsLongMonoid {  }
+
+fun JsFloat.Companion.semigroup(): JsFloatSemigroup = object : JsFloatSemigroup {  }
+
+fun JsFloat.Companion.monoid(): JsFloatMonoid = object : JsFloatMonoid {  }
+
+fun JsDouble.Companion.semigroup(): JsDoubleSemigroup = object : JsDoubleSemigroup {  }
+
+fun JsDouble.Companion.monoid(): JsDoubleMonoid = object : JsDoubleMonoid {  }
+
+fun JsString.Companion.semigroup(): JsStringSemigroup = object : JsStringSemigroup {  }
+
+fun JsString.Companion.monoid(): JsStringMonoid = object : JsStringMonoid {  }
+
+fun JsArray.Companion.semigroup(): JsArraySemigroup = object : JsArraySemigroup {  }
+
+fun JsArray.Companion.monoid(): JsArrayMonoid = object : JsArrayMonoid {  }
+
+fun JsObject.Companion.semigroup(): JsObjectSemigroup = object : JsObjectSemigroup {  }
+
+fun JsObject.Companion.monoid(): JsObjectMonoid = object : JsObjectMonoid {  }
```

Comment:
I'm missing JsNull, JsDecimal, JsNumber, and JsBoolean, also maybe we can add a Monoid of Json matching over all these monoids